src/util/openfaas/index.js

```diff
@@ -277,7 +277,7 @@ const executeFaasFunction = async (
     }
 
     if (error.statusCode === 504) {
-      throw new RespStatusError(`${name} timed out`);
+      throw new RespStatusError(`${name} timed out`, 504);
```

Comment:
status greater than 500 will be retried by server for some x number of timess. this is a definite failure, we should throw some 4XX